test/metabase/upload_test.clj

```diff
@@ -124,10 +125,30 @@
            ["My favorite number is 86"   "My favorite number is 86"   vchar-type]
            ;; Date-related
            [" 2022-01-01 "                    #t "2022-01-01"             date-type]
+           [" 2022-01-01T01:00 "              #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01t01:00 "              #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01 01:00 "              #t "2022-01-01T01:00"       datetime-type]
            [" 2022-01-01T01:00:00 "           #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01t01:00:00 "           #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01 01:00:00 "           #t "2022-01-01T01:00"       datetime-type]
            [" 2022-01-01T01:00:00.00 "        #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01t01:00:00.00 "        #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01 01:00:00.00 "        #t "2022-01-01T01:00"       datetime-type]
            [" 2022-01-01T01:00:00.000000000 " #t "2022-01-01T01:00"       datetime-type]
-           [" 2022-01-01T01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]]]
+           [" 2022-01-01t01:00:00.000000000 " #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01 01:00:00.000000000 " #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01T01:00:00.00-07 "     #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01t01:00:00.00-07 "     #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01 01:00:00.00-07 "     #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01T01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01t01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01 01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01T01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01t01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01 01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]
+           [" 2022-01-01T01:00:00.00Z "       (t/offset-date-time "2022-01-01T01:00+00:00") offset-dt-type]
```

Comment:
The `#t` reader tag will parse `"2022-01-01T01:00+00:00"` into a ZonedDateTime for some reason, so we need to use `t/offset-date-time` here